plugins/modules/omevv_firmware.py

```diff
@@ -0,0 +1,1224 @@
+                                                               cluster_group_id, **final_payload)
+        if resp.success:
+            job_resp, err_msg = self.omevv_update_obj.firmware_update_job_track(vcenter_uuid,
+                                                                                resp.json_data)
+            self.handle_job_response(parameters, vcenter_uuid, resp, job_resp, err_msg,
+                                     before_dict, after_dict)
+        else:
+            self.module.exit_json(msg=FAILED_UPDATE_MSG, failed=True, error_info=error_msg)
+        return job_resp
+
+    def handle_job_response(self, parameters, vcenter_uuid, resp, job_resp, err_msg,
+                            before_dict, after_dict):
+        """
+        Handle the response from the firmware update job.
+
+        Args:
+            parameters (dict): The parameters for the firmware update.
+            vcenter_uuid (str): The UUID of the vCenter.
+            resp (Response): The response from the firmware update.
+            job_resp (dict): The response from the firmware update job.
+            err_msg (str): The error message from the firmware update.
+            before_dict (dict): The state of the system before the update.
+            after_dict (dict): The state of the system after the update.
+
+        Returns:
+            None
+        """
+        run_now = parameters.get('run_now')
+        if not run_now:
+            self.module.exit_json(msg=SUCCESS_UPDATE_SCHEDULED_MSG, changed=True,
+                                  job_details=job_resp,
+                                  diff={"before": before_dict, "after": after_dict})
+        else:
+            job_wait = self.module.params.get('job_wait')
+            if job_wait:
+                self.wait_for_job_completion(vcenter_uuid, resp, job_resp, err_msg,
+                                             before_dict, after_dict)
+            else:
+                self.module.exit_json(msg=SUCCESS_UPDATE_SUBMIT_MSG, changed=True,
+                                      job_details=job_resp,
+                                      diff={"before": before_dict, "after": after_dict})
+
+    def wait_for_job_completion(self, vcenter_uuid, resp, job_resp, err_msg,
+                                before_dict, after_dict):
+        """
+        Waits for the firmware update job to complete.
+
+        Args:
+            vcenter_uuid (str): The UUID of the vCenter.
+            resp (Response): The response from the firmware update.
+            job_resp (dict): The response from the firmware update job.
+            err_msg (str): The error message from the firmware update.
+            before_dict (dict): The state of the system before the update.
+            after_dict (dict): The state of the system after the update.
+
+        Returns:
+            None
+
+        Raises:
+            None
+        """
+        while job_resp["state"] not in ["COMPLETED", "FAILED"]:
```

Comment:
I don't see use of job_wait_timeout here, how are we exiting when job_wait_timeout time has elapsed?